Mathlib/Data/Nat/Log.lean

```diff
@@ -319,4 +319,118 @@ theorem log_le_clog (b n : ℕ) : log b n ≤ clog b n := by
     exact (Nat.pow_le_pow_iff_right hb).1
       ((pow_log_le_self b n.succ_ne_zero).trans <| le_pow_clog hb _)
 
+/-! ### Computating the logarithm efficiently -/
+section computation
+
+private lemma logC_aux {m b : ℕ} (hb : 1 < b) (hbm : b ≤ m) : m / (b * b) < m / b := by
+  have hb' : 0 < b := zero_lt_of_lt hb
+  rw [div_lt_iff_lt_mul (Nat.mul_pos hb' hb'), ← Nat.mul_assoc, ← div_lt_iff_lt_mul hb']
+  exact (Nat.lt_mul_iff_one_lt_right (Nat.div_pos hbm hb')).2 hb
+
+-- This option is necessary because of lean4#2920
+set_option linter.unusedVariables false in
+/--
+An alternate definition for `Nat.log` which computes more efficiently. For mathematical purposes,
+use `Nat.log` instead, and see `Nat.log_eq_logC`.
+
+Note a tail-recursive version of `Nat.log` is also possible:
+```
+def logTR (b n : ℕ) : ℕ :=
+  let rec go : ℕ → ℕ → ℕ | n, acc => if h : b ≤ n ∧ 1 < b then go (n / b) (acc + 1) else acc
+  decreasing_by
+    have : n / b < n := Nat.div_lt_self (by omega) h.2
+    decreasing_trivial
+  go n 0
+```
+but performs worse for large numbers than `Nat.logC`:
+```
+#eval Nat.logTR 2 (2 ^ 1000000)
+#eval Nat.logC 2 (2 ^ 1000000)
```

Comment:
@kim-em, what's the SOTA for committing benchmarks / performance tests like these? It would be great to have a test file that computes a big logarithm, and even better if we can assert it is "quick".

Reply:
Now that I read it again, the point of this was that `Nat.logTR` times out on this computation whereas `Nat.logC` doesn't, and `Nat.log` hits stack limit at something like `Nat.log 2 (2 ^ 11000)`. 
I wouldn't mind removing these lines either, it's largely to save a future person's time if they think "I wonder if tail-recursion would be better than this weird doubling algorithm"

Reply:
My comment really was "let's add a test somewhere of something that didn't work before your change"

Reply:
I'm inclined to just create `tests/nat/log.lean`, if these are meant to be regression tests.

If they are documentation, they should be code blocks inside a module doc next to the definition.

Reply:
I don't have a strong preference here. What's the distinction between having a code block as a module doc as opposed to a code block in the definition docstring?

Reply:
> I'm inclined to just create `tests/nat/log.lean`, if these are meant to be regression tests.

We should add this file, with an invocation that took >20s before and now takes <1s.